### PR TITLE
[ci, bazel] Deprecate "vivado" tag

### PR DESCRIPTION
**Pull request by dmcardle**
This PR deletes a CI check for missing "vivado" tags, adds a new CI check that complains when it finds "vivado" tags, and removes occurrences of "vivado" from BUILD files.

Originally, this PR added some missing "vivado" tags. @timothytrippel pointed out (below) that Vivado is a system requirement now, which is corroborated by <https://github.com/lowRISC/opentitan/issues/16619#issuecomment-1396229803>.

### CODE REVIEW

**Review by timothytrippel — COMMENTED** [review]
Since we default to splicing in the freshest ROMs on each build and CI machines that build / run tests have Vivado Lab Edition installed, and Vivado lab edition being free is a OT system requirement now, I don't think we need this "vivado" tag anymore. We should probably remove it. 

IIRC I think the JTAG tag is still relied upon in CI? But the "vivado" I'm pretty sure can be removed completely.

How do you feel about updating this PR to remove these "vivado" tags completely?

**Comment by dmcardle** [comment]
@timothytrippel My goal was consistency, so deleting the "vivado" tag is even better! I'll take a stab at that tomorrow.

File: ci/scripts/check_bazel_deprecated_tags.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
```
Comment (timothytrippel) [inline ci/scripts/check_bazel_deprecated_tags.py:1]: wouldn't it be better to only check for supported tags, i.e., an allow list instead of a deny list?

**Review by timothytrippel — APPROVED** [review]

File: ci/scripts/check_bazel_deprecated_tags.py (hunk 1 above)
Comment (dmcardle) [reply ci/scripts/check_bazel_deprecated_tags.py:1]: I actually just whipped it up to demonstrate that I'd eliminated all the "vivado" tags. I intend to delete it in a follow up PR :)

Comment (dmcardle) [reply ci/scripts/check_bazel_deprecated_tags.py:1]: (Although it turns out I forgot to make the script executable. Just force pushed a new patchset and I'll let the CI run.)

**Pull request merged**